test/dart/leak_tracking/_object_tracker_test.dart

```diff
@@ -223,6 +223,16 @@ void main() {
   });
 
   group('$ObjectTracker with stack traces', () {
+    late _MockFinalizerBuilder finalizerBuilder;
+    late _MockGcCounter gcCounter;
+    late ObjectTracker tracker;
+    const disposalTimeBuffer = Duration(milliseconds: 100);
+
+    /// Emulates GC.
+    void gc(Object object) {
+      finalizerBuilder.finalizer.finalize(identityHashCode(object));
+    }
+
```

Comment:
this is all duplicated. Why can't it be defined outside of the groups so that it can be reused by both?

Reply:
I plan to create one more group that behaves differently.
So I want these groups to be independent so that the shared things can change in different directions if needed.

Reply:
If an additional group needed to create its own variables, those could still be created inside the group and then they'd be the closest in scope for the group's tests, so we should need to duplicate everywhere when only one group needs tighter scoped-variables.

Reply:
Added actual test and fix to this PR.
Reorganized the test do decrease duplicated code.

Now only three lines duplicate (declaration for finalizerBuilder, gcCounter and tracker). This is what we normally duplicate between libraries, so it should be ok to duplicate between groups.

How does it sound?

Reply:
I prefer it to be visible to the reader what mocks are used by what tests. I want tests with the same time of mocking to be grouped. It will make maintenance easier.